### PR TITLE
feat: eks/kubevela addons module rework

### PR DESCRIPTION
**Pull request by renebarbosafl**
## Summary by CodeRabbit

New Features:
- Introduced the KubeVela Controller, enhancing the system's control and management capabilities.
- Added a new Kubernetes secret named "dex-config", enhancing the system's security by securely storing configuration data.

Refactor:
- Updated the Terraform source references for EKS modules, ensuring the use of the latest and most stable versions.
- Removed the Cert-Manager section and updated the Traefik Load Balancer configuration, improving the system's load balancing capabilities.
- Removed the HTTP route TraitDefinition, indicating a shift in the system's approach to handling HTTP routes.
- Updated the configuration file for HTTPS routes, providing more customization options for gateway and listener names.
- Updated the Terraform configuration for an EKS cluster, including changes to dependencies and load balancer configuration.
- Updated the loading of environment-level variables and extraction of common variables, improving code reusability and maintainability.

### CODE REVIEW

File: infra/secrets.yaml
Comment (ce0la) [inline infra/secrets.yaml:1]: Thoughts on having a secrets directory and splitting up secrets into individual files of their own e.g. `secrets/grafana-api-key.yaml`, `secrets/dex-secret.yaml`, etc?

Comment (renebarbosafl) [reply infra/secrets.yaml:1]: I think it makes sense, @ce0la.
@tshaynik What do you think about this (Just to have a 3rd opinion)?

File: infra/modules/kubevela/main.tf
Comment (ce0la) [inline infra/modules/kubevela/main.tf:1]: Is there an alternative way to declaratively deploy Kubevela on the new cluster?

Comment (renebarbosafl) [reply infra/modules/kubevela/main.tf:1]: @ce0la This PR implements "eks-addons" submodule. This submodule is using [AWS EKS Blueprints](https://registry.terraform.io/modules/aws-ia/eks-blueprints-addons/aws/latest) to create resources on EKS (It uses helm as Terraform provider) and the code for deploying kubeVela is [here](https://github.com/input-output-hk/sc-dev-platform/blob/renebarbosa/addons-modules-rework/infra/modules/eks/addons/main.tf#L130).

So, the flow is:

1. Create EKS (Control Plane, NodeGroups, aws-auth configMap, KMS, SecurityGroups, etc);
2. Create EKS addons (cluster-autoscaler, load-balancer-controller, node-local-dns, external-dns, metrics-server and vela-core);
3. Create KubeVela Addons and resources 

For the step 3 we can use the module with declarative addons setup (Your PR).

**Review by ce0la — APPROVED** [review]

**Pull request merged**